plugins/modules/proxysql_info.py

```diff
@@ -0,0 +1,163 @@
+                - mysql_query_rules
+                - mysql_query_rules_fast_routing
+                - mysql_replication_hostgroups
+                - mysql_servers
+                - mysql_users
+                - proxysql_servers
+                - restapi_routes
+                - runtime_checksums_values
+                - runtime_global_variables
+                - runtime_mysql_aws_aurora_hostgroups
+                - runtime_mysql_firewall_whitelist_rules
+                - runtime_mysql_firewall_whitelist_sqli_fingerprints
+                - runtime_mysql_firewall_whitelist_users
+                - runtime_mysql_galera_hostgroups
+                - runtime_mysql_group_replication_hostgroups
+                - runtime_mysql_query_rules
+                - runtime_mysql_query_rules_fast_routing
+                - runtime_mysql_replication_hostgroups
+                - runtime_mysql_servers
+                - runtime_mysql_users
+                - runtime_proxysql_servers
+                - runtime_restapi_routes
+                - runtime_scheduler
+                - scheduler
+            type: list
+            returned: Always
+        global_variables:
+            description: global variables of requested proxysql
+            type: dict
+            returned: Always
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.proxysql.plugins.module_utils.mysql import mysql_connect, mysql_driver
+from ansible.module_utils.six import iteritems
+from ansible.module_utils._text import to_native
+
+# ===========================================
+# proxysql module specific support methods.
+#
+
+
+def perform_checks(module):
+    if module.params["login_port"] < 0 or module.params["login_port"] > 65535:
+        module.fail_json(msg="login_port must be a valid unix port number (0-65535)")
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            login_user=dict(default=None, type='str'),
+            login_password=dict(default=None, no_log=True, type='str'),
+            login_host=dict(default="127.0.0.1"),
+            login_unix_socket=dict(default=None),
+            login_port=dict(default=6032, type='int'),
+            config_file=dict(default="", type='path')
+        ),
+        supports_check_mode=True
+    )
+
+    perform_checks(module)
+
+    login_user = module.params["login_user"]
+    login_password = module.params["login_password"]
+    config_file = module.params["config_file"]
+
+    cursor = None
+    try:
+        cursor, db_conn, version = mysql_connect(
+            module,
+            login_user,
+            login_password,
+            config_file,
+            cursor_class='DictCursor'
+        )
+    except mysql_driver.Error as e:
+        module.fail_json(msg="unable to connect to ProxySQL Admin Module: %s" % to_native(e))
+
+    result = dict()
+    result['version'] = version
+
+    tables = list()
+    cursor.execute("show tables")
```

Comment:
```suggestion
    cursor.execute("show tables")
```
ProxySQL instances don't contain any users data (e.g. business login, etc.), right?
I'd like to be sure that there can't be gigabytes of data returned by the module (i.e. it is safe to fetch content of all available tables).

Reply:
> ProxySQL instances don't contain any users data (e.g. business login, etc.), right?

No, it contains login credentials, otherwise it does not work. To name them: `mysql_users` and `runtime_mysql_users` tables.

> I'd like to be sure that there can't be gigabytes of data returned by the module (i.e. it is safe to fetch content of all available tables).

I don't think that proxysql can contain GBs of configuration. Imo worst case we are talking about a few MBs. In Most cases < 1MB.  
You can have multiple hosts and users and maybe a lot of rewrite rules. But due the fact that more query rules slow down proxysql, it's nearly im possible that there are setups with > 100 rules. E.g. we have currently 3 rules in our production setup. 2 for R/W split and one that fixes some incompatiblities with mariadb jdbc connector.

Reply:
>>    ProxySQL instances don't contain any users data (e.g. business login, etc.), right?  
> No, it contains login credentials, otherwise it does not work. To name them: mysql_users and runtime_mysql_users tables.

I've checked, the credentials are hashed (like in mariadb/mysql/percona). So the log will not be flooded by plain passwords.

Reply:
cool, thanks for explaining!